### PR TITLE
feat: [DEL-6057]: delegate task logs for connector validation

### PR DESCRIPTION
**Pull request by abhinavrastogi-harness**
### Summary

adds ability to view delegate task logs for connector validation

#### Screenshots

Screenshot 2023-04-17 at 12 06 19 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx [hunk 1]
```diff
@@ -98,11 +94,11 @@ export function StepDetails(props: StepDetailsProps): React.ReactElement {
     return !!delegateList?.find((item: DelegateInfo) => item.taskId === taskId)
   }
 
-  const delegateLogsAvailable =
-    (step.startTs !== undefined && step.delegateInfoList && step.delegateInfoList.length > 0) ||
-    (step.startTs !== undefined && step.stepDetails && Object.keys(step.stepDetails).length > 0)
-
-  const delegateInfoList = !isEmpty(step.delegateInfoList) ? step.delegateInfoList : taskList
+  const delegateLogsAvailable = step.startTs !== undefined && step.delegateInfoList && step.delegateInfoList.length > 0
+  const timePadding = 60 * 5 // 5 minutes
```
Comment (rameshx) [inline src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx:98]: Nit: can be defined in a constants/util file and imported everywhere

**Review by rameshx — APPROVED** [review]

File: src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx [hunk 2]
```diff
@@ -98,11 +94,11 @@ export function StepDetails(props: StepDetailsProps): React.ReactElement {
     return !!delegateList?.find((item: DelegateInfo) => item.taskId === taskId)
   }
 
-  const delegateLogsAvailable =
-    (step.startTs !== undefined && step.delegateInfoList && step.delegateInfoList.length > 0) ||
-    (step.startTs !== undefined && step.stepDetails && Object.keys(step.stepDetails).length > 0)
-
-  const delegateInfoList = !isEmpty(step.delegateInfoList) ? step.delegateInfoList : taskList
+  const delegateLogsAvailable = step.startTs !== undefined && step.delegateInfoList && step.delegateInfoList.length > 0
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx:97]: Are we sure about this change? For service and environment/infrastructure steps, delegate task ids are sent as a part of `step.stepDetails`, which is why fallbacks (||) were added.

**Comment by abhinavrastogi-harness** [comment]
retrigger titlecheck

**Comment by abhinavrastogi-harness** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx [hunk 3]
```diff
@@ -222,7 +221,7 @@ export function StepDetails(props: StepDetailsProps): React.ReactElement {
                 {taskList &&
                   taskList.length > 0 &&
                   taskList.map((item, index) =>
-                    delegateListContainsTask(delegateInfoList, item.taskId) ? null : (
+                    delegateListContainsTask(step.delegateInfoList, item.taskId) ? null : (
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx:224]: Maybe, you want to give `delegateInfoList` here?

**Review by chetanvekariya-harness — DISMISSED** [review]

File: src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx (hunk 3 above)
Comment (abhinavrastogi-harness) [reply src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx]: fixed

File: src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx [hunk 4]
```diff
@@ -154,7 +153,7 @@ export function StepDetails(props: StepDetailsProps): React.ReactElement {
             </td>
           </tr>
         ))}
-        {(showDelegateRow(step.delegateInfoList, taskList) || step.stepDetails?.initStepV2DelegateTaskInfo) && (
+        {(showDelegateRow(delegateInfoList, taskList) || step.stepDetails?.initStepV2DelegateTaskInfo) && (
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx:156]: @abhinavrastogi-harness - The diff was showing that you converted `delegateInfoList` to `step.delegateInfoList`. But, in develop, I can see that it is `step.delegateInfoList` here. May be, you can revert this last change. My bad.

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**